test/native/integration-test-helpers/setup-api-fetch.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Sets up the `apiFetch` library for testing by mocking request responses.
+ *
+ * Example:
+ *
+ * const responses = [
+ *     {
+ *         request: {
+ *             path: `/wp/v2/media/1?context=edit`,
+ *         },
+ *         response: {
+ *             source_url: 'https://image-1.jpg',
+ *             id: 1,
+ *         },
+ *     },
+ *     {
+ *         request: {
+ *             path: `/wp/v2/media/2?context=edit`,
+ *         },
+ *         response: {
+ *             source_url: 'https://image-2.jpg',
+ *             id: 2,
+ *         },
+ *     },
+ * ];
+ * setupApiFetch( responses );
+ * ...
+ * expect( apiFetch ).toHaveBeenCalledWith( responses[1].request );
+ *
+ * @param {object[]} responses Array with the potential responses to return upon requests.
+ *
+ */
+export function setupApiFetch( responses ) {
+	apiFetch.mockImplementation( async ( options ) => {
+		const matchedResponse = responses.find(
+			( { request: { path, method } } ) => {
+				return (
+					path === options.path &&
+					( ! options.method || method === options.method )
+				);
+			}
+		);
+		return matchedResponse?.response;
```

Comment:
Would there be a benefit to logging an error here if there is no `matchedResponse`?

```
		if ( ! matchedResponse ) {
			throw new Error( `No response for path: ${ path }` );
		}

		return matchedResponse.response;
```

Reply:
Thank you for sharing the logic behind that decision, that makes sense to me.

> In any case, we could add an option to make this function more restrictive and fail the test if an unexpected request is made. Maybe it's useful in some cases 🤔 .

I trust your judgement, but have a sense that the value would be relatively low. The current code looks good to me :) thanks again for expanding on the logic here.

Reply:
> I trust your judgement, but have a sense that the value would be relatively low. The current code looks good to me :) thanks again for expanding on the logic here.

Ok, I'll merge the PR as-is, thanks!